contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "./GlobalAllowList.sol";
+import "./Coordinator.sol";
+import {UpfrontSubscriptionWithEncryptorsCap} from "./Subscription.sol";
+
+/**
+ * @title ManagedAllowList
+ * @notice Manages a list of addresses that are authorized to decrypt ciphertexts, with additional management features.
+ * This contract extends the GlobalAllowList contract and introduces additional management features.
+ * It maintains a reference to a Subscription contract, which is used to manage the authorization caps for different addresses and rituals.
+ * The Subscription contract is used to enforce limits on the number of authorization actions that can be performed, and these limits can be set and updated through the ManagedAllowList contract.
+ */
+contract ManagedAllowList is GlobalAllowList {
+    mapping(bytes32 => uint256) internal allowance;
+
+    /**
+     * @notice The Subscription contract used to manage authorization caps
+     */
+    UpfrontSubscriptionWithEncryptorsCap public subscription;
+
+    /**
+     * @notice Emitted when an administrator cap is set
+     * @param ritualId The ID of the ritual
+     * @param _address The address of the administrator
+     * @param cap The cap value
+     */
+    event AdministratorCapSet(uint32 indexed ritualId, address indexed _address, uint256 cap);
+
+    /**
+     * @notice Sets the coordinator and subscription contracts
+     * @dev The coordinator and subscription contracts cannot be zero addresses
+     * @param _coordinator The address of the coordinator contract
+     * @param _subscription The address of the subscription contract
+     */
+    constructor(
+        Coordinator _coordinator,
+        UpfrontSubscriptionWithEncryptorsCap _subscription
+    ) GlobalAllowList(_coordinator) {
+        require(address(_coordinator) != address(0), "Coordinator cannot be the zero address");
+        require(address(_subscription) != address(0), "Subscription cannot be the zero address");
+        subscription = _subscription;
```

Comment:
I do not see any method to change subscription, maybe we can make it `immutable`?